client/src/components/categorical/category/index.tsx

```diff
@@ -641,13 +643,56 @@ const CategoryValueList = React.memo(
     colorData,
     colorTable,
     colorMode,
+    isCellGuideCxg,
   }: CategoryValueListProps) => {
-    const tuples = [...categorySummary.categoryValueIndices];
+    let tuples = [...categorySummary.categoryValueIndices];
+    if (isCellGuideCxg) {
+      tuples = tuples.filter(
+        ([, index]) => categorySummary.categoryValueCounts[index] > 0
+      );
 
-    /*
-    Render the value list.  If this is a user annotation, we use a flipper
-    animation, if read-only, we don't bother and save a few bits of perf.
-    */
+      if (
+        colorMode === "color by continuous metadata" ||
+        colorMode === "color by expression" ||
+        colorMode === "color by geneset mean expression"
+      ) {
+        const categoryDataArray = categoryData.col(metadataField).asArray();
+        const colorDataArray = colorData.icol(0).asArray();
+        const categoryColorMap = new Map();
+        categoryDataArray.forEach((category: string, index: number) => {
+          if (!categoryColorMap.has(category)) {
+            categoryColorMap.set(category, { sum: 0, count: 0 });
+          }
+          const colorValue = colorDataArray[index];
+          const categoryColor = categoryColorMap.get(category);
+          categoryColor.sum += colorValue;
+          categoryColor.count += 1;
+        });
+
+        const categoryAverageColor = new Map();
+        categoryColorMap.forEach((value, key) => {
+          categoryAverageColor.set(key, value.sum / value.count);
+        });
+        tuples.sort((a, b) => {
+          const colorA = categoryAverageColor.get(a[0]);
+          const colorB = categoryAverageColor.get(b[0]);
+          return colorB - colorA;
+        });
+      }
+      // (alec): This is to sort the categories by the number of cells in each category
+      // Disabling for now, but it could be useful.
+      // else {
+      //   tuples.sort(
+      //     (a, b) =>
+      //       categorySummary.categoryValueCounts[b[1]] -
+      //       categorySummary.categoryValueCounts[a[1]]
+      //   );
+      // }
+      /*
+      Render the value list.  If this is a user annotation, we use a flipper
+      animation, if read-only, we don't bother and save a few bits of perf.
+      */
```

Comment:
if we don't expect to re-enable this code, let's remove it since the code is saved in git tree already